File: Programming_Languages/Java/Binary_Search.java
import java.util.Scanner;
class BinarySearch {

	int binarySearch(int arr[], int l,int r, int x)
	{
		if (r >= l && l<arr.length-1) {
		
			int mid = l + (r - l) / 2;
			if (arr[mid] == x)
				return mid;

			if (arr[mid] > x)
				return binarySearch(arr, l, mid - 1, x);
			return binarySearch(arr, mid + 1, r, x);
		}

		return -1;
	}
	public static void main(String args[])
	{
		BinarySearch ob = new BinarySearch();
        System.out.println("Enter number of elements : "); 
        Scanner s=new Scanner(System.in);
        int n = s.nextInt();
        int arr[] = new int[n];

        System.out.println("Enter elements in ascendibg order: ");
        for(int i=0; i < n; i++){  
                arr[i] = s.nextInt(); 
        }   
        
        System.out.println("Enter element to be searched : ");
		int x = s.nextInt();
		int result = ob.binarySearch(arr, 0, n - 1, x);
		if (result == -1)
			System.out.println("Element not present");
		else
			System.out.println("Element found at index " + result);
        s.close();
	}
}

File: Programming_Languages/C/Binary Search.c
#include<stdio.h>
int main()
{
	int key,mid,i,j,t;
	int lb=0;
	int ub=10;
	int arr[10]={45,28,10,1,65,56,7,20,41,98};
	printf("My initial array:-\n");
	for(i=0;i<10;i++)
	printf("%d ",arr[i]);
	//Sorting array using bubble sort
	for(i=0;i<10;i++)
	{
		for(j=0;j<9-i;j++)
		{
			if(arr[j]>arr[j+1])
			{
				t=arr[j];
				arr[j]=arr[j+1];
				arr[j+1]=t;
			}
		}
	}
	printf("\nAfter sorting the array:-\n");
	for(i=0;i<10;i++)
	printf("%d ",arr[i]);
	//Binary search
	printf("\nPlease enter the element to search:-");
	scanf("%d",&key);
	mid=(lb+ub)/2;
	while(arr[mid]!=key && lb<=ub)
	{
		if(arr[mid]<key)
		lb=mid+1;
		else
		ub=mid-1;
		mid=(lb+ub)/2;
	}
	if(arr[mid]==key)
	printf("your element is found at index %d",mid+1);
	else
	printf("Element not found in the array");
	return 0;
}
